lib/improver/tests/nbhood/circular_kernel/test_CircularPercentiles.py

```diff
@@ -0,0 +1,396 @@
+        self.assertIsInstance(result, Cube)
+        self.assertIsInstance(result.coord(
+            'percentiles_over_neighbourhood'), iris.coords.Coord)
+        self.assertArrayEqual(result.coord(
+            'percentiles_over_neighbourhood').points,
+            PercentileConverter.DEFAULT_PERCENTILES)
+
+
+class Test_run(IrisTest):
+
+    """Test neighbourhood circular percentile plugin."""
+
+    def test_basic(self):
+        """Test that the plugin returns an iris.cube.Cube."""
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_time_points=1,
+            num_grid_points=5)
+        radius = 4000.
+        result = CircularPercentiles().run(cube, radius)
+        self.assertIsInstance(result, Cube)
+
+    def test_single_point(self):
+        """Test behaviour for a single non-zero grid cell."""
+        cube = set_up_cube()
+        expected = np.ones([15] + list(np.shape(cube.data)))
+        expected = np.transpose(expected, [1, 0, 2, 3, 4])
+        expected[0, :, 0, 7, 5] = PERCENTILES_1_IN_13
+        expected[0, :, 0, 6:9, 6] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 5:10, 7] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (5, 1))))
+        expected[0, :, 0, 6:9, 8] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 7, 9] = PERCENTILES_1_IN_13
+        radius = 4000.
+        result = CircularPercentiles().run(cube, radius)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_multi_point_multitimes(self):
+        """Test behaviour for points over multiple times."""
+        cube = set_up_cube(
+            zero_point_indices=[(0, 0, 10, 10), (0, 1, 7, 7)],
+            num_time_points=2
+        )
+        expected = np.ones([15] + list(np.shape(cube.data)))
+        expected = np.transpose(expected, [1, 0, 2, 3, 4])
+        expected[0, :, 0, 10, 8] = PERCENTILES_1_IN_13
+        expected[0, :, 0, 9:12, 9] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 8:13, 10] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (5, 1))))
+        expected[0, :, 0, 9:12, 11] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 10, 12] = PERCENTILES_1_IN_13
+        expected[0, :, 1, 7, 5] = PERCENTILES_1_IN_13
+        expected[0, :, 1, 6:9, 6] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 1, 5:10, 7] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (5, 1))))
+        expected[0, :, 1, 6:9, 8] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 1, 7, 9] = PERCENTILES_1_IN_13
+        radius = 4000.
+        result = (CircularPercentiles().run(cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_lat_long(self):
```

Comment:
I'm not sure this tests will be required as we now have a separate utility for this checking.